clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -741,6 +742,12 @@ void CIRGenModule::replaceGlobal(mlir::cir::GlobalOp Old,
   // If the types does not match, update all references to Old to the new type.
   auto OldTy = Old.getSymType();
   auto NewTy = New.getSymType();
+  auto OldAS = Old.getAddrSpaceAttr();
+  auto NewAS = New.getAddrSpaceAttr();
+  // TODO(cir): If the AS differs, we should also update all references.
+  if (OldAS != NewAS) {
+    llvm_unreachable("NYI");
+  }
```

Comment:
NYI because we won't meet such situation any soon.

Reply:
Sounds good!